src/main/java/com/flowingcode/vaadin/addons/gridexporter/ExcelInputStreamFactory.java

```diff
@@ -174,41 +187,101 @@ private void buildRow(T item, Sheet sheet, Cell startingCell) {
     currentColumn[0] = startingCell.getColumnIndex();
     exporter.columns.forEach(column -> {
       Object value = exporter.extractValueFromColumn(item, column);
+      value = transformToType(value, column);
       Cell currentCell = startingCell;
       if (startingCell.getColumnIndex() < currentColumn[0]) {
         currentCell = startingCell.getRow().createCell(currentColumn[0]);
-        currentCell.setCellStyle(startingCell.getCellStyle());
+        CellStyle newCellStyle = currentCell.getSheet().getWorkbook().createCellStyle();
+        newCellStyle.cloneStyleFrom(startingCell.getCellStyle());
+        currentCell.setCellStyle(newCellStyle);
+
+        configureAlignment(column.getTextAlign(), currentCell);
       }
       currentColumn[0] = currentColumn[0] + 1;
-      buildCell(value, currentCell);
-
+      buildCell(value, currentCell, column);
     });
   }
 
+  private Object transformToType(Object value, Column<T> column) {
+    Object result = value;
+    if (value instanceof String) {
+      String stringValue = (String) value;
+      String parsingFormatPattern =
+          (String) ComponentUtil.getData(column, GridExporter.COLUMN_PARSING_FORMAT_PATTERN_DATA);
+      try {
+        if (parsingFormatPattern != null) {
+          switch ((String) ComponentUtil.getData(column, GridExporter.COLUMN_TYPE_DATA)) {
+            case GridExporter.COLUMN_TYPE_NUMBER:
+              DecimalFormat decimalFormat = new DecimalFormat(parsingFormatPattern);
```

Comment:
This parser instance uses the default locale, but it has likely been formatted using either the UI locale or a custom locale (plus symbols).

Reply:
Yeah, one alternative would be to provide the `DecimalFormat` instead of the pattern, what do you think?